code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -629,22 +645,29 @@
 				icon_state = "alarm_b1"
 		return
 
-	if((stat & (NOPOWER|BROKEN)) || shorted)
-		icon_state = "alarmp"
-		return
-
+	icon_state = "alarm1"
+	var/overlay_state = AALARM_OVERLAY_OFF
 	var/area/A = get_area(src)
 	switch(max(danger_level, A.atmosalm))
 		if(0)
-			icon_state = "alarm0"
+			overlay_state = AALARM_OVERLAY_GREEN
+			light_color = LIGHT_COLOR_GREEN
+			set_light(brightness_on)
 		if(1)
-			icon_state = "alarm2" //yes, alarm2 is yellow alarm
+			overlay_state = AALARM_OVERLAY_WARN
+			light_color = LIGHT_COLOR_LAVA
+			set_light(brightness_on)
 		if(2)
-			icon_state = "alarm1"
+			overlay_state = AALARM_OVERLAY_DANGER
+			light_color = LIGHT_COLOR_RED
+			set_light(brightness_on)
+
+	SSvis_overlays.add_vis_overlay(src, icon, overlay_state, ABOVE_LIGHTING_LAYER, ABOVE_LIGHTING_PLANE, dir)
+	update_light()
 
 /obj/machinery/airalarm/process()
 	if((stat & (NOPOWER|BROKEN)) || shorted)
-		return
+		return FALSE
```

Comment:
process doesn't have a return value it cares about except `PROCESS_KILL` so there's no reason to add one here